src/category_theory/sites/cocontinuous.lean

```diff
@@ -0,0 +1,201 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import category_theory.sites.sheaf
+import category_theory.limits.kan_extension
+
+/-!
+# Cocontinuous functors between sites.
+
+We define cocontinuous functors between sites as functors that pulls covering sieves back to
+covering sieves. This seems stronger that the definition found in stacks project,
+but they are actually equivalent via `category_theory.grothendieck_topology.superset_covering`
+(The precise statement not formalized, but follows from it quite trivially).
+
+This concept is also called the "covering lifting property" as defined in [MM92] Chapter VII,
+Section 10, and should not be confused with the general definition of cocontinuous functors
+between categories as functors preserving small colimits.
+
+## Main Results
+- `category_theory.sites.cocontinuous.Ran_is_sheaf`: If `F : C ⥤ D` is cocontinuous, then
+`Ran F.op` as a functor `(Cᵒᵖ ⥤ A) ⥤ (Dᵒᵖ ⥤ A)` of presheaves maps sheaves to sheaves.
+
+## References
+
+* [S. MacLane, I. Moerdijk, *Sheaves in Geometry and Logic*][MM92]
+* https://stacks.math.columbia.edu/tag/00XI
+
+-/
+
+universes u
+noncomputable theory
+
+open category_theory
+open opposite
+open category_theory.presieve.family_of_elements
+open category_theory.presieve
+open category_theory.limits
+
+namespace category_theory
+namespace sites
+variables {C D A : Type u} [category.{u} C] [category.{u} D] [category.{u} A] [has_limits A]
+variables {J : grothendieck_topology C} {K : grothendieck_topology D}
+
+/--
+A functor `F : (C, J) ⥤ (D, K)` between sites is called cocontinuous if for all covering sieves
+`R` in `D`, `R.pullback F` is a covering sieve in `C`.
+-/
+structure cocontinuous (J : grothendieck_topology C) (K : grothendieck_topology D) (F : C ⥤ D) :=
+(cover_lift : ∀ {U : C} {S : sieve (F.obj U)} (hS : S ∈ K (F.obj U)), S.functor_pullback F ∈ J U)
+
+namespace cocontinuous
+
+/-- A trivial example to make the inhabited linter happy. -/
+instance id_cocontinuous : inhabited (cocontinuous J J (𝟭 _)) :=
+⟨⟨λ _ S h, by { convert h, simp }⟩⟩
+
+
+variables {F : C ⥤ D} (HF : cocontinuous J K F)
+namespace Ran_is_sheaf
+/-
+We will now prove that `Ran F.op` maps sheaves to sheaves if `F` is cocontinuous. This can be found
+in https://stacks.math.columbia.edu/tag/00XK. However, the proof given here uses the
+amalgamation definition of sheaves, and thus does not require that `C` or `D` has catigorical
+pullbacks.
+
+Suppose there is a compatible family `x` of elements on `U`, we ought to glue them together
+to form a unique section on `S`. We will achieve this by restricting `x` onto objects of the form
+`F(Y)` with `Y : C`, and glueing them via the sheaf property of `ℱ` on `C`.
+The uniqueness also follows from the uniqueness provided by the sheaf property.
+-/
+variable (ℱ : Sheaf J A)
```

Comment:
I find it slightly confusing that you're using `F` for the functor and `ℱ` for the sheaf. Maybe rename the former into something else, like `G`?

Reply:
Yeah sure.
For me the `ℱ` looks too different to be called an F and I just take it as a general unpronounceable symbol that represents some sheaf XD.